### DISCUSSION TITLE
Get global reference frame using the LightHouse Positioning system

### DISCUSSION THREAD

**Discussion by alexandre-bonnefond** [bitcraze/discussions#1113] [Q&A] [answered]
Hello everyone, 

I've just bought the Lighthouse positioning system and I plan to do some swarming with interactions between the crazyflies. 

For that application, I mostly need the distance between the crazyflies but I struggle to have the position of each one in a global frame. Each crazyflie seems to have its "origin" where it was initialized regarding the Lighthouse. This totally makes sense but does not really help me for my goal. 

So I wanted to know if I should precisely initialize all the crazyflies at the exact same location or is there a way to fix a "virtual" origin with the lighthouse and then use it as a reference origin for all the instances ? 

Thank you in advance for your help !

**Comment by knmcguire** [comment, marked as answer]
Hi!

The lighthouse is a global positioning system, so there 'should' be only one point of origin for all. It seems to me in this case that for each crazyflie, you have done the [geometry estimation in the lighthouse tab](https://www.bitcraze.io/documentation/tutorials/getting-started-with-lighthouse/#estimate-the-geometry) separately? 

For a swarm, it's better to only to the basestation geometry for only **one** crazyflie (which is placed on where you'd like the (0,0) to be, then do:
1. 'Save system config' to save that geometry .yaml file
2. connect to another crazyflie
3. 'Load system config' and select that same .yaml file you used for the very first crazyflie
4. Repeat 2-3 for each crazyflie using only the one .yaml file

This way, all crazyflies have a shared similar geometry of the base station and therefore should all have the same origin in their coordinate system in space. 

Remember that this geometry of the basestations is saved in the eeprom memory of the crazyflie, which is persistent even when they restart. You don't have to do the estimation or load the geometry file with every startup.

**Reply by alexandre-bonnefond** [reply]
Hi ! 

Thank you that answers my question !

**Reply by knmcguire** [reply]
ah great! Perhaps we should clarify this bit at the end of the tutorial :)
